[서인덕] Week2

Dev-Duke-Seo opened the pull request:
## 요구사항

### 기본
- [X] [기본] Github에 1 ~ 2주차 위클리 미션 Pull Request를 만들었나요?
- [X] [기본] Netlify에 파일 배포가 아닌 포크한 Github 레포지토리로 연결했나요?
- [X] [랜딩 페이지] 아래로 스크롤 해도 “Linkbrary” 로고와 “로그인” 버튼이 있는 상단 네비게이션 바(Global Navigation Bar)영역이 최상단에 고정되게 했나요?
- [X] [로그인 페이지, 회원가입 페이지 공통] Linkbrary 로고 클릭시 루트 페이지(“/”)로 이동하나요?
- [X] [로그인 페이지, 회원가입 페이지 공통] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일한가요?
- [X] [로그인 페이지, 회원가입 페이지 공통] input 요소에 focus in 일 때, 테두리 색상이 파랑색인가요?
- [X] [로그인 페이지, 회원가입 페이지 공통] input 요소에 focus out 일 때, 테두리 색상이 회색인가요?
- [X] [로그인 페이지, 회원가입 페이지 공통] SNS 아이콘들은 클릭시 각각 “https://www.google.com/”, “https://www.kakaocorp.com/page/” 으로 이동하나요?
- [X] [로그인 페이지] “회원 가입하기”는 클릭시 “/signup” 페이지로 이동하나요?
- [X] [회원가입 페이지] “로그인 하기”는 클릭시 “/signin” 페이지로 이동하나요?

###  심화
- [X] [심화] palette에 있는 color값들을 css 변수로 등록하고 사용했나요?
- [X] [심화] 비밀번호 input 요소 위에 비밀번호를 확인할 수 있는 눈 모양 아이콘을 추가했나요?

## 주요 변경사항

- index, signin, signup 페이지 구현완료하였습니다.
- 각 페이지에 대한 요구사항/기능에 대한 테스트를 완료하였습니다.

## 스크린샷
![screenshot_signin](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/127701344/80f8d0a9-e427-439f-bde0-1aec2baff05e)
![screenshot_signup](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/assets/127701344/2968c49a-5f85-46f7-a0e9-9275dd2e21ab)

## 멘토에게
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

Dev-Duke-Seo committed:
first commit

Dev-Duke-Seo committed:
feat: Make HTML pages and Stylesheets

1) created index.html with index.css
2) created signup.html and signin.html with sign.css

Dev-Duke-Seo committed:
feat: Make HTML pages and Stylesheets

1) created index.html with index.css
2) created signup.html and signin.html with sign.css

Dev-Duke-Seo committed:
Add head-meta tags

1) Add meta tags for Facebook and Twitter
   -> most of Twitter meta-tags are substituted by OG

Dev-Duke-Seo committed:
Add head-meta tags

1) Add meta tags for Facebook and Twitter
   -> most of Twitter meta-tags are substituted by OG

Dev-Duke-Seo committed:
Add head-meta tags

1) Add meta tags for Facebook and Twitter
   -> most of Twitter meta-tags are substituted by OG

Dev-Duke-Seo committed:
Add head-meta tags

1) Add meta tags for Facebook and Twitter
   -> most of Twitter meta-tags are substituted by OG

Dev-Duke-Seo committed:
feat: Add head-meta tags

1) Add meta tags for Facebook and Twitter
   -> most of Twitter meta-tags are substituted by OG

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  /*gap: 8px;*/
```
주석은 왠만하면 삭제 해주세요. 의미 있는 주석은 언제든지 달아야 하지만 이것처럼 안쓰게 돼서 적은것같은 주석은 추후에 유지보수에 혼동을 야기합니다!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+html {
+  font-family: Pretendard, Abel, sans-serif;
+  word-break: keep-all;
+}
```
word-break라는 속성을 html태그에 주게되면 해당 파일내에 모든 요소에  word-break: keep-all;이 먹힐거에요. 만약 디자이너분과 협의해서 줄바꿈이 안일어나는게 default라면 상관없지만 해당 요소는 일반적이진 않아서 저는 조금 위험하다고 생각해요.

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+}
+
+header {
+  z-index: 1;
```
z-index 주신거 정말 좋습니다! 이렇게 직접 숫자로 넣어줘도 좋지만 팔레트에 추후 z-index에 대한 레이어를 추상화하여 해당 값을 쓰는것도 좋습니다! 
--zIndex-least-value: 1;

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  height: 24px;
+}
+
+#login {
```
해당 html 확인해봤는데 버튼이더라구요. id값은 하나밖에 존재할 수 없는 unique한 값이기 때문에 section과 같은 유니크한 값에 적용해주는것이 좋습니다. 하지만 버튼은 그것과는 조금 거리가 있다고 생각해서 loginButton같은 classname을 사용하시거나 css 명시도로 해결하시는걸 추천드려요!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  align-items: center;
+  gap: 10px;
+  border-radius: 8px;
+  background: var(--gra-purpleblue-to-skyblue);
```
저는 backgroung-color를 줄때는 background보다는 더 명확한 요소로 주는 편이에요. 왜냐하면 background는 color말고도 다른 요소가 있어서 해당 프로퍼티를 읽을떄는 이 요소엔 왜 color만 있지라는 의문이 생기더라구요,
```suggestion
  background-color: var(--gra-purpleblue-to-skyblue);
```

Dev-Duke-Seo replied:
주신 코멘트대로 색상을 background 대신, background-color로 대체해서 적용해보려했는데 안 되더라구요! 

조사해보니 ,linear-gradient() 함수는 두 개 이상의 색이 직선을 따라 점진적으로 변화하는 이미지를 생성한다고합니다.  -> 적용하려는 배경색상이 linear-gradient(91deg, #6D6AFE 0.12%, #6AE3FE 101.84%)를 변수화한 컬러여서 색상속성으로는 적용이 불가했습니다.

endmoseung replied:
아하 확인했습니다! gradient이외에는 backgroundColor로 부탁드릴게요~

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  justify-content: flex-end;
+  align-items: center;
+  gap: 40px;
+  background: #F0F6FF;
```
여기두!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  font-size: 64px;
+  font-style: normal;
+  font-weight: 700;
+  line-height: 80px; /* 125% */
```
주석이 의미있는값이 아니라면 지워주세요:)

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+}
+
+.text-gradation-type-head {
+  background: linear-gradient(91deg, #6D6AFE 17.28%, #FF9F9F 74.98%);
+  background-clip: text;
+  -webkit-background-clip: text;
+  -webkit-text-fill-color: transparent;
+}
+
+.text-gradation-type-a {
+  background: linear-gradient(96deg, #FE8A8A 1.72%, #A4CEFF 74.97%);
+  background-clip: text;
+  -webkit-background-clip: text;
+  -webkit-text-fill-color: transparent;
+}
+
+.text-gradation-type-b {
+  background: linear-gradient(277deg, #6FBAFF 59.22%, #FFD88B 93.66%);
+  background-clip: text;
+  -webkit-background-clip: text;
+  -webkit-text-fill-color: transparent;
+}
+
+.text-gradation-type-c {
+  background: linear-gradient(99deg, #6D7CCD 19.76%, rgba(82, 136, 133, 0.22) 52.69%);
+  background-clip: text;
```
오오 이렇게 추상화해둔것 너무 좋아요 다음번에 재사용하기 좋겠네요

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  align-items: center;
+  gap: 10px;
+  border-radius: 8px;
+  background: var(--gra-purpleblue-to-skyblue);
```
여기두!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+
+  border-radius: 25px;
+  background: linear-gradient(180deg, rgba(196, 196, 196, 0.00) 67.68%, #F0F6FF 94.76%);
+  /*box-shadow: 0 4px 25px 0 rgba(0, 0, 0, 0.08);*/
```
주석은 지워주세요!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 400;
+  line-height: 150%; /* 24px */
```
여기도!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  width: 550px;
+  height: 450px;
+  border-radius: 15px;
+  background: #F0F6FF;
```
여기두!

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  position: relative;
+  width: 550px;
+  height: 450px;
+  border-radius: 15px;
```
이렇게 4로떨어지지 않는 값은 주로 디자이너의 실수일 확률이 높아요. 제 말이 100퍼신뢰되는건 아니지만 현업에서는 주로 4, 8의 배수로 디자인 작업하는 편이에요 관련 링크 첨부할게요:) https://brunch.co.kr/@bf6b5403fa344c4/2

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+}
+
+
+/*------------------------------------------------*/
```
이 부분도 좋은것 같아요 푸터를 분리해두시려고 작성하신거죠 ?

endmoseung commented on index.css:
```diff
@@ -0,0 +1,311 @@
+  border-radius: 15px;
+  /*background: linear-gradient(180deg, rgba(255, 255, 255, 0.00) 48.72%, #000 128.17%);*/
+  /*box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.08);*/
+}
```
여기도 지워주세요!

endmoseung commented on index.html:
```diff
@@ -0,0 +1,127 @@
+</head>
+<body>
+<header>
+  <nav>
```
상단 navbar는 nav태그써주는것이 일반적입니다 굿!!

endmoseung commented on index.html:
```diff
@@ -0,0 +1,127 @@
+    </article>
+  </section>
+</main>
+
```
이렇게 한 줄 구분하는것도 좋은 컨벤션이라고 생각합니다:)

endmoseung commented on index.html:
```diff
@@ -0,0 +1,127 @@
+          FAQ
+        </p>
+      </div>
+      <div class="icons">
```
저만의 코드 컨벤션 규칙이 있는데 비슷한 자식들을 2개 이상 감싸고 있으면 container, 단순 하나의 자식을 감싸고 있으면 wrapper 예를들어 여기서는 class="SNS-icons-container"이런식으로 지을 수도 있을것 같아요. classname이 길면 코드가 길어보이는거 같은 단점이 있지만 의도를 전하기엔 명확해서 두개의 장단점을 비교해보는것도 좋을것 같습니다

endmoseung commented on sign.css:
```diff
@@ -0,0 +1,215 @@
+    justify-content: center;
+    align-items: center;
+    gap: 10px;
+
```
👍

endmoseung commented on signin.html:
```diff
@@ -0,0 +1,63 @@
+        </div>
+    </div>
+
+    <form class="login-form" action="/submit_form" method="GET">
```
👍

endmoseung reviewed (approved):
확인했습니다! 인덕님 정말 고생하셨어요:)
전체적으로 신경쓰신 부분이 많이 보이고 잘하신게 티가 났던것 같습니다!
부족하기보다는 좀 더 개선될 수 있는 부분에 대해서 리뷰 남겼습니다.

- 커밋을 기능별로 나누셨으면 좋겠어요 주로 png와 같이 정적인 파일을 올릴때는 저는 chore라는 커밋 헤더르 붙이는 편이에요. ex) chore: 회원가입 페이지에 필요한 이미지들 추가
- 실제로 작업하신게 있고 netlify로 배포하셨다면 해당 링크도 공유주셨으면 좋겠어요. 제가 코드를 리뷰하는 입장에서 어떤 코드가 있었는지도 궁금하지만 해당 배포사이트에서도 확인할 수 있기에 하신건 다 공유주시면 감사하겠습니다!
- 어떤 부분을 신경쓰셨는지도 적어주시면 좋은 리뷰문화가 생길 것 같아요. 가령 인덕님이 이번에 flex를 다루는 부분에 신경쓰셨고 해당 내용을 적어주시면 제가 그 내용위주로 좀 더 디테일하게 리뷰를 남길 수도있을것 같아요.
- 주요 변경사항도 적어주시면 감사하겠습니다! 커밋을 보면 다 알수 있긴하지만 코드리뷰의 베이스는 상대방에 대한 배려라고 생각하기에 어떤 변경사항이 생겼는지 적어주어 리뷰가 더 잘 될 수 있도록 하면 좋을것 같습니다:)

전체적으로 아쉬운부분은 건호님과 비슷하여 공통으로 남길게요:)

Dev-Duke-Seo commented:
@endmoseung 
멘토님 정성스런 코멘트 감사합니다! 
1. netlify에 배포된 사이트 주소 공유드립니다.
->  https://sprint-test-duke.netlify.app 
**  작업내용을 반영/배포 하기 위한 프로덕션 용으로 개인 별도 repo에 연결되어있습니다. 
2. 짚어주신 포인트들 잘 확인하여 refactor 해두겠습니다.
3. 외에 주신 좋은 커멘트들 숙지하여 이후 작업에 반영하겠습니다 : )

Dev-Duke-Seo committed:
Refactor: Improve code structure for better readability

- Update naming of Some class or id
- Apply abstraction to improve code readability
- Resolve icon-box position error

Dev-Duke-Seo committed:
Refactor/Fix: 기능 추가 및 코드 구조 개선

- refactor: index.html / css
1) 컬러팔레트 별도 파일로 분리 및 import 기능 활용

- fix: index.html
1) faq, privacy 페이지 누락 추가.
2) 클릭 가능한 링크에 대한 pointer 기능 누락 추가
3) footer 영역 snsicon에 대한 새창에서 링크 열기 추가.
4) header / main영역 1920px에 대한 배경 채움효과 개선

Dev-Duke-Seo committed:
Refactor/Fix: 기능 추가 및 코드 구조 개선

- refactor: index.html / css
1) 컬러팔레트 별도 파일로 분리 및 import 기능 활용

- fix: index.html
1) faq, privacy 페이지 누락 추가.
2) 클릭 가능한 링크에 대한 pointer 기능 누락 추가
3) footer 영역 snsicon에 대한 새창에서 링크 열기 추가.
4) header / main영역 1920px에 대한 배경 채움효과 개선

Dev-Duke-Seo committed:
Feat: index//반응형 디자인(Tablet size)구현

1199px 이하의 화면에서의 style 변경.

Dev-Duke-Seo committed:
Feat: index//반응형 디자인(Mobile size)구현

767px 이하의 화면에서의 style 변경.

Dev-Duke-Seo committed:
Feat: sign-in&sign-up//반응형 디자인(Mobile size)구현

- 767px 이하의 화면에서의 style 변경.
- 기존 일부 오류 수정

Dev-Duke-Seo committed:
fix: Footer 영역 사이즈 반응오류 수정

- index.css // .footer-area의 padding 오류 수정.

Dev-Duke-Seo commented:
# 반응형 웹사이트 제작 / 메타태그 적용
## 1. 기본
- [x]  브라우저에 현재 보이는 화면의 영역(viewport) 너비를 기준으로 분기되는 반응형 디자인을 적용했나요?
- [x]   페이스북, 카카오톡, 디스코드, 트위터 등 SNS에서 Linkbrary 랜딩 페이지(“/”) 공유 시 미리보기를 볼 수 있도록 랜딩 페이지 메타 태그를 설정했나요?
- [x]   미리보기에서 제목은 “Linkbrary”, 설명은 “세상의 모든 정보를 쉽게 저장하고 관리해 보세요”로 설정했나요?
- [x]   주소와 이미지는 자유롭게 설정했나요?
- [x]   Tablet 사이즈에서 화면의 너비가 1199px 이하로 작아질 때 “Linkbrary” 로고와 “로그인” 버튼 사이의 간격은 변하지 않게 고정값으로 유지하되 좌우 여백이 줄어드나요?
- [x]   Tablet 사이즈에서 최소 좌우 여백은 “Linkbrary” 로고의 왼쪽에 여백 32px, “로그인” 버튼 오른쪽 여백 32px을 유지하고“Linkbrary” 로고와 “로그인" 버튼의 간격이 가까워지나요?
- [x]   PC, Tablet 사이즈의 이미지 크기는 고정값을 사용했나요?
- [x]   Mobile 사이즈의 이미지는 좌우 여백 32px을 제외하고 이미지 영역이 꽉 차나요?
- [x]   Mobile 사이즈 너비가 커지면, “Privacy Policy”, “FAQ”, “codeit-2023”이 있는 영역과 SNS 아이콘들이 있는 영역의 간격이 커지나요?
- [x]   Tablet 사이즈에서 디자인은 PC사이즈와 동일한가요?
- [x]   Mobile 사이즈에서 좌우 여백 32px 제외하고 내부 요소들이 너비를 모두 차지하나요?
- [x]   Mobile 사이즈에서 내부 요소들의 너비는 기기의 너비가 커지는 만큼 커지지만 400px을 넘지 않나요?
## 심화
- [ ]   팀원의 위클리 미션 PR에 코드 리뷰를 남겼나요?
- [ ]   랜딩 페이지 Mobile 사이즈에서 제품 소개 영역의 순서를 제목, 설명, 이미지 => 제목, 이미지, 설명 순서로 변경했나요?

## 주요 변경사항
- pc, 태블릿, 모바일 버전에서 다른 CSS 적용
- 메타태그 적용

## Comment
- Deployed on : https://sprint-test-duke.netlify.app/ 
@endmoseung 
- 멘토님! 지난번 리뷰 이후 머지되지 않은 상태에서 포크된 레포지토리로 계속 푸쉬를했더니 금일 머지될 때 함께 머지되었습니다.  -> 이후 머지이전 작업 내용들은 모두 새로운 브랜치로 분기하여 작업 해두겠습니다.

endmoseung commented on index.css:
```diff
@@ -11,7 +11,8 @@
   --gra-purpleblue-to-skyblue: linear-gradient(91deg, #6D6AFE 0.12%, #6AE3FE 101.84%);
-  --fandom-k-gra-blue-to-pink: linear-gradient(271deg, #FE578F -9.84%, #68E8F9 107.18%)
+  --fandom-k-gra-blue-to-pink: linear-gradient(271deg, #FE578F -9.84%, #68E8F9 107.18%);
+  --zIndex-least-value: 1;
```
좋습니다! 이런식으로 zIndex도 계층을 정리해두면 좋은게 프로그램이 복잡해질수록 다양한 레이어가 쌓여서 UI가 그려지는데 그때 기준을 위의 zIndex로 얘기할 수 있어서 편리해집니다:)

endmoseung commented on index.css:
```diff
@@ -50,7 +33,7 @@ header {
 
 header nav {
-
+  max-width: 1920px;
```
이 화면은 FHD PC유저까지만 적용할거야! 라면 크게 상관없지만 요즘 QHD까지 사용하는 유저가 많아서 이부분은 설정하지 않아도 괜찮을것 같습니다

endmoseung commented on index.css:
```diff
@@ -84,16 +67,30 @@ nav .button {
   line-height: normal;
 }
 
+main {
```
전역 css에 이렇게 태그에 스타일 먹이는건 괜찮으나 추후 전역 css에서 특정 태그에 대한 스타일은 기본적으로 적용해야하는것 말고는 설정하지 않는게 좋습니다!

endmoseung commented on index.css:
```diff
@@ -312,4 +313,10 @@ footer {
+a[href] {
+  cursor: pointer;
+}
+
```
이 커밋에 대한 내용을 까먹을거 같아서 여기 작성해요! Refactor/Fix: 기능 추가 및 코드 구조 개선 이 커밋명처럼 명확하게 기능이 구분되면 그냥 커밋을 두개로 나누던가 아 이정도면 굳이 커밋 나눌필요 없다고 하면 그대로 하셔도 괜찮을것 같습니다!

endmoseung commented on index.html:
```diff
@@ -1,152 +1,169 @@
+  <meta property="og:title" content="Linkbrary">
+  <meta property="og:description" content="세상의 모든 정보를 쉽게 저장하고 관리해보세요.">
+  <meta property="og:image" content="resources/featured_libraries.jpg">
+  <meta property="og:url" content="/index.html">
```
이부분은 배포한 사이트에 URL을 넣으셔야 될 것 같아요!

endmoseung commented on index.html:
```diff
@@ -1,152 +1,169 @@
+  <meta name="twitter:card" content="summary_large_image">
+  <meta name="twitter:image" content="resources/featured_libraries.jpg">
+  <!--  <meta name="twitter:title" content="Linkbrary">-->
+  <!--  <meta name="twitter:description" content="세상의 모든 정보를 쉽게 저장하고 관리해보세요.">-->
```
안쓰는 주석은 제거 부탁드려요~

endmoseung commented on index.css:
```diff
@@ -397,4 +393,167 @@ a[href] {
   }
+}
+
+/*-------Responsive: Mobile---------*/
```
사람마다 호불호가 있는데 저는 이렇게 구분해주는 주석이 좋더라구요:)

endmoseung commented on index.css:
```diff
@@ -397,4 +393,167 @@ a[href] {
+
+  .text-subject .mobile-invisible {
+    display: none;
+  }
```
비슷한 요소로 visiblillty라는 속성이 있는데 display로 none을 적용하면 html태그가 실제로 화면에 적용되는 과정에서 빠지기때문에 아에 paint하지 않지만 visibillty는 단순히 보이는부분만 없애주는 효과가 있어요 나중에 쓰일일 있으면 적용해보세요:)

endmoseung reviewed (commented):
확인했습니다 인덕님! 고생하셨어요.
커밋 잘게 나누신거 너무 잘 된것 같습니다. 커밋마다 명확하게 어떤일을 하신지 알 수 있어서 리뷰남기는데도 너무 좋았네요👍
크게 아쉬운점은 없고 리뷰 남겼으니 추후 반영한 부분 커밋까지 해서 4주차에 올려주세요:)
